src/nvim/ex_docmd.c

```diff
@@ -8411,8 +8421,11 @@ static void ex_redir(exarg_T *eap)
 /// ":redraw": force redraw
 static void ex_redraw(exarg_T *eap)
 {
-  if (State & CMDPREVIEW) {
-    return;  // Ignore :redraw during 'inccommand' preview. #9777
+  // Ignore :redraw during 'inccommand' preview. (#9777)
+  // Except when executing the command which is being previewed,
+  // because previewing requires a redraw.
+  if ((State & CMDPREVIEW) && !(State & PREVIEW_EXEC)) {
```

Comment:
can we check the `preview_arg` or something here instead of another special global state flag?

Reply:
> can we check the `preview_arg` or something here instead of another special global state flag?

Checking `preview_arg` won't be possible, but I suppose we *could* use a global variable instead of a state flag

Reply:
> I suppose we could use a global variable instead of a state flag

The idea is to avoid extra global flags of any kind :)

Reply:
> > I suppose we could use a global variable instead of a state flag
> 
> The idea is to avoid extra global flags of any kind :)

I understand, I don't like global variables either. But I'm not sure how we can check things in this case without them. At least I can't think of a way to do that myself.